### PR TITLE
Add gradient accumulation doc

### PR DESCRIPTION
**Pull request by muellerzr**
Adds a doc for gradient accumulation for the next release. It's a carbon copy of the one in #505.

### CODE REVIEW

File: docs/source/gradient_accumulation.mdx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    inputs, targets = batch
+-    inputs = inputs.to(device)
+-    targets = targets.to(device)
+    outputs = model(inputs)
+    loss = loss_function(outputs, targets)
+    loss = loss / gradient_accumulation_steps
++    accelerator.backward(loss)
+    if (index+1) % gradient_accumulation_steps == 0:
+        optimizer.step()
+        scheduler.step()
```
Comment (sgugger) [inline docs/source/gradient_accumulation.mdx:70]: Make sure to add a consistent indent level of 2 for the base, so that you can have `+ ` or `- ` without changing indentation on any line.

File: docs/source/gradient_accumulation.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+You just wrap it around the entire training part of our code: 
+
+```diff
+-for index, batch in enumerate(training_dataloader):
++for batch in training_dataloader:
++   with accelerator.accumulate(model):
+        optimizer.zero_grad()
+        inputs, targets = batch
+        outputs = model(inputs)
+```
```
Comment (sgugger) [inline docs/source/gradient_accumulation.mdx:98]: Same here.

File: docs/source/gradient_accumulation.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        scheduler.step()
+```
+
+<Tip warning={true}>
+In its current state, this code is not going to perform gradient accumulation efficiently due to a process called gradient synchronization. Read more about that in the Concepts tutorial!
+</Tip>
+
+## Letting 🤗 Accelerate handle gradient accumulation
+
+All that is left now is to let 🤗 Accelerate handle the gradient accumulation for us. To do so you should pass in a `gradient_accumulation_steps` parameter to `Accelerator`, dictating the number
```
Comment (sgugger) [inline docs/source/gradient_accumulation.mdx:79]:
```suggestion
All that is left now is to let 🤗 Accelerate handle the gradient accumulation for us. To do so you should pass in a `gradient_accumulation_steps` parameter to [`Accelerator`], dictating the number 
```

File: docs/source/gradient_accumulation.mdx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+and you can remove all the special checks for the step number and the loss adjustment:
+
+```diff
+-loss = loss / gradient_accumulation_steps
+accelerator.backward(loss)
+-if (index+1) % gradient_accumulation_steps == 0:
+optimizer.step()
+scheduler.step()
+```
```
Comment (sgugger) [inline docs/source/gradient_accumulation.mdx:108]: Same there + the step lines change since they lose and indent level.

**Review by sgugger — APPROVED** [review]
Thanks for adding this!
